technologydata/sources.py

```diff
@@ -208,6 +222,378 @@ def process(self, trusted_execution: bool = False) -> None:
+        else:
+            return None
+
+    @staticmethod
+    def _get_content_type(url_archived: str) -> Any:
+        """
+        Fetch the content type of the archived URL.
+
+        This method sends a HEAD request to the specified archived URL to retrieve the
+        Content-Type from the response headers. It returns the content type as a string
+        if the request is successful; otherwise, it logs an error and returns None.
+
+        Parameters
+        ----------
+        url_archived : str
+            The URL of the archived resource for which the content type is to be fetched.
+
+        Returns
+        -------
+        str | None
+            The Content-Type of the archived URL if the request is successful, or None
+            if an error occurs during the request.
+
+        Raises
+        ------
+        requests.exceptions.RequestException
+            If there is an issue with the HTTP request, an error is logged, and None is returned.
+
+        """
+        try:
+            response = requests.head(url_archived)
+            response.raise_for_status()
+            return response.headers.get("Content-Type")
+        except requests.exceptions.RequestException as e:
+            logger.error(f"Failed to retrieve content type: {e}")
+            return None
+
+    @staticmethod
+    def _download_file(
+        url_archived: str, save_path: pathlib.Path
+    ) -> pathlib.Path | None:
+        """
+        Download the file and save it to the specified path.
+
+        This method retrieves the content from the specified archived URL and saves it
+        to the provided file path. It handles HTTP errors and logs appropriate messages
+        based on the outcome of the download operation.
+
+        Parameters
+        ----------
+        url_archived : str
+            The URL of the archived file to be downloaded.
+
+        save_path : pathlib.Path
+            The path where the downloaded file will be saved, including the file name.
+
+        Returns
+        -------
+        pathlib.Path | None
+            The path where the file has been saved if the download is successful, or None
+            if an error occurs during the download process.
+
+        Raises
+        ------
+        requests.exceptions.RequestException
+            If there is an issue with the HTTP request, an error is logged, and None is returned.
+
+        """
+        try:
+            response = requests.get(url_archived)
+            response.raise_for_status()  # Check for HTTP errors
+
+            with open(save_path, "wb") as file:
+                file.write(response.content)
+
+            logger.info(f"File downloaded successfully and saved to {save_path}")
+            return save_path
+        except requests.exceptions.RequestException as e:
+            logger.error(f"An error occurred during file download: {e}")
+            return None
+
+    @staticmethod
+    def is_wayback_snapshot_available(
```

Comment:
We don't use the method and we don't need it, because `savepagenow.capture_or_cache(...)` provides this information already.

Reply:
Less code :rocket: